src/dust/coagulation/coagulation.cpp

```diff
@@ -0,0 +1,308 @@
+namespace Coagulation {
+//----------------------------------------------------------------------------------------
+//! \fn  StateDescriptor Coagulalation::Initialize
+//! \brief Adds intialization function for coagulation package
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto coag = std::make_shared<StateDescriptor>("coagulation");
+  Params &params = coag->AllParams();
+
+  CoagParams cpars;
+  const int nm = pin->GetOrAddInteger("dust", "nspecies", 1);
+  cpars.nm = nm;
+  cpars.vfrag = pin->GetOrAddReal("dust", "vfrag", 1.e3); // cm/s
+  cpars.nlim = 1e10;
+  cpars.integrator = pin->GetOrAddInteger("dust", "coag_int", 3);
+  cpars.use_adaptive = pin->GetOrAddBoolean("dust", "coag_use_adaptiveStep", true);
+  cpars.mom_coag = pin->GetOrAddBoolean("dust", "coag_mom_preserve", true);
+  cpars.nCall_mx = pin->GetOrAddInteger("dust", "coag_nsteps_mx", 1000);
+  // dust particle internal density g/cc
+  const Real rho_p = pin->GetOrAddReal("dust", "grain_density", 1.25);
+  cpars.rho_p = rho_p;
+
+  cpars.ibounce = pin->GetOrAddBoolean("dust", "coag_bounce", false);
+
+  int coord_type = 0; // density
+
+  const bool isurface_den = pin->GetOrAddBoolean("dust", "surface_density_flag", true);
+  if (isurface_den) coord_type = 1;
+  cpars.coord = coord_type; // 1--surface density, 0: 3D
+
+  cpars.err_eps = 1.0e-1;
+  cpars.S = 0.9;
+  cpars.cfl = 1.0e-1;
+  cpars.chi = 1.0;
+
+  // some checks and definition
+  if (cpars.use_adaptive) {
+    if (cpars.integrator == 3) {
+      cpars.pgrow = -0.5;
+      cpars.pshrink = -1.0;
+    } else if (cpars.integrator == 5) {
+      cpars.pgrow = -0.2;
+      cpars.pshrink = -0.25;
+    } else {
+      std::stringstream msg;
+      msg << "### FATAL ERROR in dust coagulation initialization: " << std::endl
+          << "###   You can not use this integrator with adaptive step sizing: "
+          << cpars.integrator << std::endl;
+      PARTHENON_FAIL(msg);
+    }
+    cpars.errcon = std::pow((5. / cpars.S), (1. / cpars.pgrow));
+  }
+
+  const Real s_min = pin->GetReal("dust", "min_size");
+  const Real s_max = pin->GetReal("dust", "max_size");
+  const Real mmin = 4.0 * M_PI / 3.0 * rho_p * std::pow(s_min, 3);
+  const Real mmax = 4.0 * M_PI / 3.0 * rho_p * std::pow(s_max, 3);
+  const Real cond = 1.0 / (1.0 - nm) * std::log(mmin / mmax);
+  const Real conc = std::log(mmin);
+  if (std::exp(cond) > std::sqrt(2.0)) {
+    std::stringstream msg;
+    msg << "### FATAL ERROR in dust with coagulation: using nspecies >"
+        << std::log(mmax / mmin) / (std::log(std::sqrt(2.0))) + 1. << " instead of " << nm
+        << std::endl;
+    PARTHENON_FAIL(msg);
+  }
+
+  ParArray1D<Real> dust_size("dustSize", nm);
+  auto dust_size_host = dust_size.GetHostMirror();
+
+  for (int n = 0; n < nm; ++n) {
+    Real mgrid = std::exp(conc + cond * n);
+    dust_size_host(n) = std::pow(3.0 * mgrid / (4.0 * M_PI * cpars.rho_p), 1. / 3.);
+  }
+  dust_size.DeepCopy(dust_size_host);
```

Comment:
The dust package has already read an initialized the dust size array, so this can be removed

Reply:
fixed in the new updated code